be/src/exec/schema_scanner.cpp

```diff
@@ -233,11 +233,7 @@ Status SchemaScanner::_create_slot_descs(ObjectPool* pool) {
 
     for (int i = 0; i < _column_num; ++i) {
         TSlotDescriptor t_slot_desc;
-        auto type_desc = TypeDescriptor(_columns[i].type);
-        if (_columns[i].type == LogicalType::TYPE_VARCHAR || _columns[i].type == LogicalType::TYPE_CHAR) {
-            type_desc.len = _columns[i].size;
-        }
-
+        TypeDescriptor type_desc = _columns[i].type;
```

Comment:
```suggestion
        const TypeDescriptor& type_desc = _columns[i].type;
```
to avoid unnecessary copying.

Reply:
done